### PR TITLE
[query] Use CodeBuilder interface for buildSkip

### PR DESCRIPTION
**Pull request by chrisvittal**

### CODE REVIEW

**Comment by chrisvittal** [comment]
stacked on #9371

**Comment by chrisvittal** [comment]
ready for review, of course it's failing the tests right now.

**Comment by chrisvittal** [comment]
Tests pass. Probably messed up my translation the first time.

File: hail/src/main/scala/is/hail/types/encoded/ETransposedArrayOfStructs.scala [hunk 1]
```diff
@@ -158,57 +158,42 @@ final case class ETransposedArrayOfStructs(
...
     val prefix = Code(len := in.readInt(), nMissing := UnsafeUtils.packBitsToBytes(len))
     if (structRequired) {
-      Code(
-        prefix,
-        alen := len,
-        anMissing := UnsafeUtils.packBitsToBytes(alen),
-        fmbytes := r.allocate(const(1), nMissing.toL),
-        skipFields
-      )
+      cb.assign(alen, len)
```
Comment (patrick-schultz) [inline hail/src/main/scala/is/hail/types/encoded/ETransposedArrayOfStructs.scala:173]: Previous version had `anMissing := UnsafeUtils.packBitsToBytes(alen)` here. Was that removed intentionally? It looks to me like that could be factored out of the if branches to the `fields.foreach`. And the `fmbytes` assignment as well.

**Pull request merged**